### DISCUSSION TITLE
SSL Certificate add failed, Error: 1312 A specified logon session does not exist. It may already have been terminated.

### DISCUSSION THREAD

**Discussion by mike-credisense** [datalust/seq-tickets#2110] [Help] [answered]
I'm trying to bind the ssl thumb print as per the windows instructions with the bind-ssl statement with the thumbprint of my wild card certificate that was installed as per the instructions for the windows server deployment https/ssl documentation

**SSL Certificate add failed, Error: 1312
A specified logon session does not exist. It may already have been terminated.
Failed to bind the certificate: `netsh` returned exit code 1.**

How can we resolve this to enable https communication?

**Comment by liammclennan** [comment]
Hi @mike-credisense,

The most common reasons for this error are:

 • The certificate isn't imported into the correct store (needs to be Local Machine/Personal), or
 • One of the certs in its certificate chain is not trusted by Windows on the server (need to add the root certificate to the trusted root certificates store), or
 • The certificate has been imported without its private key (this generally means it wasn't exported in full, with its private key, to begin with - reimporting with the private key will fix this).

If none of those look promising there are a bunch more possibilities on [this stackoverflow question](https://stackoverflow.com/questions/13076915/ssl-certificate-add-failed-when-binding-to-port).

**Reply by mike-credisense** [reply]
thanks for that got it to bind, based on the second two points in your list

However no when the config is change to https://<dns name>, https:<dns name>:5341  along wit the ingestion port being set to 5341 when i start the service it stops shortly after, only seems to want to work with http://localhost:5341 
eg the following comand line entirs <domain name> redacted

`seq config set -k api.listenUris -v https://seq.<domain name>,https://seq.<domain name>:5341
seq config set -k api.ingestionPort -v 5341
seq service restart`

We have applied the license before doing this, and checked netstat for conflicts

**Reply by liammclennan** [reply]
Hi @mike-credisense 

What command did you use to bind the SSL thumbprint?

Also check the log files in the `/Log` directory in your Seq data directory. If the service is stopping there will be an error in the log.

**Reply by mike-credisense** [reply]
seq bind-ssl --thumbprint="<thumb print from certificate>"

**Reply by mike-credisense** [reply]
the documentation mentions -- port but i'm happy for it to run on the standard https port  443, and it;/s a single instance server

**Reply by mike-credisense** [reply]
No logs so I checked the config file an updated the logging path and get a log now that makes it easier to see whats going on

**Preregister URL prefixes on the server** 

very handy the log gives the command to run I had to add prefix url for both end points for web and ingestion port

```
nnetsh http add urlacl url=https://seq.<domain url>:443/ user=<servername>\<Username>
nnetsh http add urlacl url=https://seq.<domain url>:<ingestion port#>/ user=<servername>\<Username>
```

All starts up nicely now and front end is accessible after firewall  entries for 443 and ingestion port#.

My issue now is that the web url accepts the ingestion but the ingestion port does not, both have like firewall rules set up the only difference being the port number, addign Sreilog logging shows 

The SSL connection could not be established
Connect forcibly closed by host

**Comment by mike-credisense** [comment, marked as answer]
Solution

1) Import the private key to the personal folder and trusted root certificates folder
2) register the domains to be used
```
nnetsh http add urlacl url=https://seq.<domain url>:443/ user=<servername>\<Username>

nnetsh http add urlacl url=https://seq.<domain url>:<ingestion port#>/ user=<servername>\<Username>
```
credit to Liam in answers above

**Reply by liammclennan** [reply]
Hi @mike-credisense 

Glad you found a solution, and thank you for letting us know.
